### PR TITLE
Snippet highlighting based on hover

### PR DESCRIPTION
**Pull request by sean-purcell**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -33,7 +33,8 @@ function LessonInteractiveArea(lesson: ILesson) {
   const exercisesListItems = [];
   for (const exercise of lesson.exercises) {
     exercisesListItems.push(
-      <Snippet.SnippetFrame {...exercise.content} />
+      <Snippet.Snippet {...exercise.content} />
+      // <Snippet.SnippetFrame {...exercise.content} />
```
Comment (zhyty) [inline src/App.tsx]: Do we still need this comment here?

File: src/Snippet.tsx [hunk 2]
```diff
@@ -19,4 +22,51 @@ function SnippetFrame(code: ISnippet) {
...
+    const classes = langclass + " block";
+    for (let i = 0; i < this.props.blocks.length; i++) {
+      const blkclss = classes + " " + highlightCssClass(this.state.highlight[i]);
+      rb.push(<div className={blkclss}>
+        <code className={blkclss}>{this.props.blocks[i]}</code>
+      </div>)
+    }
+    return (
+      <pre className={langclass}>
+        {...rb}
```
Comment (zhyty) [inline src/Snippet.tsx]: I think {rb} suffices. There's no need to spread the array here.

File: src/Snippet.tsx [hunk 3]
```diff
@@ -19,4 +22,51 @@ function SnippetFrame(code: ISnippet) {
...
+
+  public render() {
+    const rb = [];
+    const langclass = "language-" + this.props.lang;
+    const classes = langclass + " block";
+    for (let i = 0; i < this.props.blocks.length; i++) {
+      const blkclss = classes + " " + highlightCssClass(this.state.highlight[i]);
+      rb.push(<div className={blkclss}>
+        <code className={blkclss}>{this.props.blocks[i]}</code>
+      </div>)
```
Comment (zhyty) [inline src/Snippet.tsx]: That's weird. There's no semicolon here?

**Review by zhyty — CHANGES_REQUESTED** [review]
Looks good. Just small things I'm wondering about. Lemme know if this kinda review is too nit-picky for this project.

File: src/App.tsx (hunk 1 above)
Comment (sean-purcell) [reply src/App.tsx]: Was planning on doing clean pass when this feature is closer to complete

File: src/Snippet.tsx (hunk 2 above)
Comment (sean-purcell) [reply src/Snippet.tsx]: Ok, will that just concatenate the elements?

File: src/Snippet.tsx (hunk 3 above)
Comment (sean-purcell) [reply src/Snippet.tsx]: oops

**Comment by sean-purcell** [comment]
The `<div>` thing broke indentation/newlines, don't merge yet

File: src/Snippet.tsx (hunk 2 above)
Comment (zhyty) [reply src/Snippet.tsx]: Yeah, unless I'm misunderstanding something. [This](https://github.com/schmubleck/yiddish-cornstarch/blob/master/src/App.tsx#L43) works the same way.

File: src/Snippet.tsx (hunk 3 above)
Comment (zhyty) [reply src/Snippet.tsx]: how did you beat tslint 😮

**Review by zhyty — APPROVED** [review]

**Pull request merged**